File: 53_Maximum_Subarray/xsy/solution.go
func maxSubArray(nums []int) int {
    max := nums[0]
    current := -1
    for _, i := range nums {
        if current < 0 {
            current = i
        } else {
            current += i
        }
        if current > max {
            max = current
        }
    }
    return max
}

File: 53_Maximum_Subarray/xsy/solution.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max = nums[0]
        current = -1
        for i in nums:
            if current < 0:
                current = i
            else:
                current += i
            if current > max:
                max = current
        return max 
        